test/common/performance_counter_test.cpp

```diff
@@ -0,0 +1,136 @@
+#include <functional>
+#include <random>
+#include <vector>
+#include "common/json.h"
+#include "gtest/gtest.h"
+#include "util/multi_threaded_test_util.h"
+#include "util/performance_counter_test_util.h"
+
+namespace terrier {
+
+/**
+ * Helper class for testing the four attributes of a CacheCounter.
+ */
+class CacheCounterTestObject {
+ private:
+  std::atomic<uint64_t> num_insert{0};
+  std::atomic<uint32_t> num_hit{0};
+  std::atomic<uint16_t> num_failure{0};
+  std::atomic<uint8_t> num_user{0};
+  CacheCounter *cc;
+
+  /**
+   * Calls the gtest EXPECT_EQ function for all the relevant variables.
+   */
+  void Equal() const {
+    EXPECT_EQ(cc->Get_num_insert(), num_insert);
+    EXPECT_EQ(cc->Get_num_hit(), num_hit);
+    EXPECT_EQ(cc->Get_num_failure(), num_failure);
+    EXPECT_EQ(cc->Get_num_user(), num_user);
+  }
+
+  std::vector<double> work_probs = std::vector<double>(8, 0.125);
+  std::vector<std::function<void()>> workloads = {
```

Comment:
auto should be fine here?

Reply:
error: 'auto' not allowed in non-static class member